bundles/plugins/org.bonitasoft.studio.validators/src/org/bonitasoft/studio/validators/provider/ValidatorBarResourceProvider.java

```diff
@@ -81,13 +83,22 @@ public void addResourcesForConfiguration(final BusinessArchiveBuilder builder, f
                 }
             }
         }
+        return resources;
+    }
 
-        for (final BarResource barResource : resources) {
-            builder.addExternalResource(barResource);
+    protected SourceFileStore findSourceFileStoreForValidator(final ValidatorSourceRepositorySotre validatorSourceStore, final String validatorId,
+            final ValidatorDescriptor descriptor) {
+        final AbstractFileStore child = validatorSourceStore.getChild(descriptor.getClassName());
+        if (child == null) {
+            throw new RuntimeException("Validator class " + descriptor.getClassName() + " not found for validator definition " + validatorId + "!");
+        }
+        if (!(child instanceof SourceFileStore)) {
+            throw new RuntimeException("Invalid validator definition " + validatorId + "!");
```

Comment:
we don't really know if it is because the implementation is not a valid java class or not. it is the present case but it can be everything